game/state/city/vequipment.cpp

```diff
@@ -89,22 +101,27 @@ bool VEquipment::fire(GameState &state, Vec3<float> targetPosition, Vec3<float>
 
 	auto fromScaled = vehicleMuzzle * VELOCITY_SCALE_CITY;
 	auto toScaled = targetPosition * VELOCITY_SCALE_CITY;
-	City::accuracyAlgorithmCity(state, fromScaled, toScaled, type->accuracy + owner->getAccuracy(),
-	                            targetVehicle && targetVehicle->isCloaked());
 
-	Vec3<float> velocity = toScaled - fromScaled;
-	velocity = glm::normalize(velocity);
-	// I believe this is the correct formula
-	velocity *= type->speed * PROJECTILE_VELOCITY_MULTIPLIER;
+	for (number_of_shots; number_of_shots > 0; number_of_shots--)
```

Comment:
There's no need for the first statement in this for loop, it could just be:
for (;number_of_shots > 0; number_of_shots--)
{....}
But then you could just do:
while (number_of_shots--)
{....}

Or it might be more explicit to do:

for (int shot_number = 0; shot_number < number_of_shots; shot_number++)
{....}

or similar - adding new variables isn't bad, especially if those variables are well named after what they represent (free documentation :)

Reply:
Taking your comment below into account also, would this be better for this entire section of code?

(Also consider my reluctance to change from a check on null/0 value for number_of_shots in <burst> due to the aforementioned issues )

`for (int shot_number = 0; shot_number < number_of_shots; shot_number++)
	{
		City::accuracyAlgorithmCity(state, fromScaled, toScaled,
		                            type->accuracy + owner->getAccuracy(),
		                            targetVehicle && targetVehicle->isCloaked());

		Vec3<float> velocity = toScaled - fromScaled;
		velocity = glm::normalize(velocity);
		// I believe this is the correct formula
		velocity *= type->speed * PROJECTILE_VELOCITY_MULTIPLIER;

		const auto projectile_type = type->isGuided ? Projectile::Type::Missile : Projectile::Type::Beam; owner, targetVehicle,
		    homingPosition, muzzle, velocity, type->turn_rate, type->ttl, type->damage, /*delay*/ 0,
		    /*depletion rate*/ 0, type->tail_size, type->projectile_sprites, type->impact_sfx,
		    type->explosion_graphic, state.city_common_image_list->projectileVoxelMap,
		    type->stunTicks, type->splitIntoTypes, manual);
		owner->tileObject->map.addObjectToMap(projectile);
		owner->city->projectiles.insert(projectile);
	}`

Reply:
Yes, I think that looks a bit clearer - that's probably the "most common" for loop construct style, so most experienced c++ coders will instantly recognise what the loop is doing - and optimising for "developer understanding" is a good goal.